modules/olm-creating-fb-index-image.adoc

```diff
@@ -0,0 +1,153 @@
+//
+// * operators/admin/olm-managing-custom-catalogs.adoc
+
+ifdef::openshift-origin[]
+:registry-image: quay.io/openshift/origin-operator-registry:4.9.0
+endif::[]
+ifndef::openshift-origin[]
+:registry-image: registry.redhat.io/openshift4/ose-operator-registry:v4.9
+endif::[]
+
+[id="olm-creating-fb-index-image_{context}"]
+= Creating a file-based index image
+
+You can create an index image that uses a plain text _file-based catalog_ format (JSON or YAML) instead of the deprecated SQLite database format. The `opm` CLI has new tooling that helps initialize an index in the file-based format, render new records into it, and then validate that the index is valid.
+
+.Prerequisites
+
+* `opm` version 1.18+
+* `podman` version 1.9.3+
+* A bundle image built and pushed to a registry that supports link:https://docs.docker.com/registry/spec/manifest-v2-2/[Docker v2-2]
+
+.Procedure
+
+. Initialize an index for a file-based catalog:
+
+.. Create a directory for the index:
++
+[source,terminal]
+----
+$ mkdir <operator_name>-index
+----
+
+.. In the index directory, create a Dockerfile that can build an index image:
++
+.Example `<operator_name>-index.Dockerfile`
+[source,bash,subs="attributes+"]
+----
+# The base image is expected to contain
+# /bin/opm (with a serve subcommand) and /bin/grpc_health_probe
+FROM {registry-image}
+
+# Configure the entrypoint and command
+ENTRYPOINT ["/bin/opm"]
+CMD ["serve", "/configs"]
+
+# Copy declarative config root into image at /configs
+ADD <operator_name>-index /configs
+
+# Set DC-specific label for the location of the DC root directory
+# in the image
+LABEL operators.operatorframework.io.index.configs.v1=/configs
+----
+
+.. Populate the index with your package definition:
++
+[source,terminal]
+----
+$ opm init <operator_name> \ <.>
+    --default-channel=preview \ <.>
+    --description=./README.md \ <.>
+    --icon=./operator-icon.svg \ <.>
+    --output yaml \ <.>
+    > <operator_name>-index/index.yaml <.>
+----
+<.> Operator, or package, name.
+<.> Channel that subscription will default to if unspecified.
+<.> Path to the Operator's `README.md` or other documentation.
+<.> Path to the Operator's icon.
+<.> Output format: JSON or YAML.
+<.> Path for creating the index configuration file.
++
+This command generates an `olm.package` declarative config blob in the specified index configuration file.
+
+. Add a bundle to the index:
++
+[source,terminal]
+----
+$ opm render <registry>/<namespace>/<bundle_image_name>:<tag> \ <.>
+    --output=yaml \
+    >> <operator_name>-index/index.yaml <.>
+----
+<.> Pull spec for the bundle image.
+<.> Path to the index configuration file.
++
+The `opm render` command generates a declarative config blob from the provided index images and bundle images.
++
+[NOTE]
+====
+Channels must contain at least one bundle.
+====
+
+. Add a channel entry for the bundle. For example, modify the following to add to your index configuration file:
```

Comment:
Suggest: Add a channel entry for the bundle. For example, modify the following to add to your index configuration file(<operator_name>-index/index.yaml):